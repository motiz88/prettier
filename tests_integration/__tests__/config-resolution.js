"use strict";

const path = require("path");

const runPrettier = require("../runPrettier");
const prettier = require("../../");

test("resolves configuration from external files", () => {
  const output = runPrettier("cli/config/", ["**/*.js"]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});

test("resolves configuration from external files and overrides by extname", () => {
  const output = runPrettier("cli/config/", ["**/*.ts"]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});

test("accepts configuration from --config", () => {
  const output = runPrettier("cli/config/", [
    "--config",
    ".prettierrc",
    "./js/file.js"
  ]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});

test("resolves configuration file with --find-config-path file", () => {
  const output = runPrettier("cli/config/", [
    "--find-config-path",
    "no-config/file.js"
  ]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});

test("prints nothing when no file found with --find-config-path", () => {
  const output = runPrettier("cli/config/", ["--find-config-path", ".."]);
  expect(output.stdout).toEqual("");
  expect(output.status).toEqual(1);
});

test("CLI overrides take precedence", () => {
  const output = runPrettier("cli/config/", ["--print-width", "1", "**/*.js"]);
  expect(output.stdout).toMatchSnapshot();
  expect(output.status).toEqual(0);
});

test("API resolveConfig with no args", () => {
  return prettier.resolveConfig().then(result => {
    expect(result).toBeNull();
  });
});

test("API resolveConfig with file arg", () => {
  const file = path.resolve(path.join(__dirname, "../cli/config/js/file.js"));
  return prettier.resolveConfig(file).then(result => {
    expect(result).toMatchObject({
      tabWidth: 8
    });
  });
});

test("API resolveConfig with file arg and extension override", () => {
  const file = path.resolve(
    path.join(__dirname, "../cli/config/no-config/file.ts")
  );
  return prettier.resolveConfig(file).then(result => {
    expect(result).toMatchObject({
      semi: true
    });
  });
});
