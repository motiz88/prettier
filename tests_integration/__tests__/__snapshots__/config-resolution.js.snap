// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI overrides take precedence 1`] = `
"/* eslint-disable */
console.log(
  \\"should have semi\\"
);
/* eslint-disable */
console.log(
  \\"should not have semi\\"
)
/* eslint-disable */
console.log(
  \\"should have semi\\"
);
/* eslint-disable */

function f() {
        console.log(
                \\"should have tab width 8\\"
        );
}
\\"use strict\\";

module.exports = {
        tabWidth: 8
};
/* eslint-disable */

function f() {
  console.log(
    \\"should have no semicolons\\"
  )
}
/* eslint-disable */

function f() {
   console.log(
      \\"should have tab width 3\\"
   );
}
/* eslint-disable */

function f() {
  console.log.apply(
    null,
    [
      'this file',
      'should have trailing comma',
      'and single quotes',
    ],
  );
}
"
`;

exports[`accepts configuration from --config 1`] = `
"/* eslint-disable */

function f() {
  console.log(\\"should have tab width 8\\")
}
"
`;

exports[`resolves configuration file with --find-config-path file 1`] = `
".prettierrc
"
`;

exports[`resolves configuration from external files 1`] = `
"/* eslint-disable */
console.log(\\"should have semi\\");
/* eslint-disable */
console.log(\\"should not have semi\\")
/* eslint-disable */
console.log(\\"should have semi\\");
/* eslint-disable */

function f() {
        console.log(\\"should have tab width 8\\");
}
\\"use strict\\";

module.exports = {
        tabWidth: 8
};
/* eslint-disable */

function f() {
  console.log(\\"should have no semicolons\\")
}
/* eslint-disable */

function f() {
   console.log(\\"should have tab width 3\\");
}
/* eslint-disable */

function f() {
  console.log.apply(null, [
    'this file',
    'should have trailing comma',
    'and single quotes',
  ]);
}
"
`;

exports[`resolves configuration from external files and overrides by extname 1`] = `
"function g() {
  console.log(\\"should have semicolons because it has a .ts extension\\");
}
function g() {
     console.log(\\"should have tab width 5 because it has .ts extension\\");
}
"
`;
